### PR TITLE
Resource Return Endpoint

### LINKED ISSUE #26: Add endpoint for returning a borrowed resource
We will need an endpoint to update when a resource no longer needed and the borrower returns it to the owner.

### PR DESCRIPTION
**Pull request by Bentheburrito**
Adds an endpoint for a borrower to return a resource. Updated the API outline to reflect these changes, and tested success/error responses locally in Postman.

Bonus: added the `type` filter for `/api/item/` to the API outline

Closes #26

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -339,6 +339,64 @@ app.post('/api/item/cancel-reservation/:id', async (req, res) => {
...
+    } else if (resource.reserved_by != user_id) { // The user cancelling the reservation has to be the borrower
+      return res.status(401).send({
+        error: API_RETURN_MESSAGES.UNAUTHORIZED,
+        message: "You are not authorized to perform this action."
+      });
+    }
+
+    const _update_result = await sql`
+      update resources
+      set reservation_status = ${ITEM.UNLISTED}, reserved_by = NULL where id = ${id};
```
Comment (diegovega49) [inline index.js:382]: Looks good but is there a reason the item is set as unlisted after it's returned?

Comment (Bentheburrito) [reply index.js:382]: I figured we shouldn't assume the owner wants to lend it out again, though maybe a `relist` bool query param would be better?

Comment (diegovega49) [reply index.js:382]: That sounds good, but you have a point. Listing/unlisting should be at the owner's discretion. Maybe that should be a separate feature?

Comment (Bentheburrito) [reply index.js:382]: True, I don't think we have an endpoint specifically for listing and unlisting resources right now - probably worth creating an issue for that

**Review by diegovega49 — APPROVED** [review]

**Pull request merged**